### PR TITLE
Fix window float enable/disable (#1960)

### LINKED ISSUE #1620: Glitches are shown when resizing the window (Linux)
This is forked from #1618 

Video: https://github.com/hajimehoshi/ebiten/issues/1618#issuecomment-825659074 (The window froze in the video, but freezing has already been fixed)

CC @sedyh

### PR DESCRIPTION
**Pull request by eliasdaler**
When resizing with i3 (and possibly other WMs), the only way to now new framebuffer/window size is to get event from glfw.
This also calls `setWindowSizeInDIP` to update window size afterwards (instead of calling `context.Layout` directly)

Framebuffer size callbacks are used for handling window resize in glfw examples [like here](https://github.com/glfw/glfw/blob/master/examples/gears.c#L248)

I'm not 100% sure that this is the best solution, so any suggestions welcome.
@sedyh this might also fix #1620 - can you please check (there are still glitches, but maybe it's somewhat better)?

### CODE REVIEW

File: internal/uidriver/glfw/ui.go [hunk 1]
```diff
@@ -821,7 +822,13 @@ event:
 			time.Sleep(time.Millisecond)
 		}
 	}
-	window.SetFramebufferSizeCallback(glfw.ToFramebufferSizeCallback(nil))
+	if u.defaultFramebufferSizeCallback == 0 {
+		u.defaultFramebufferSizeCallback = glfw.ToFramebufferSizeCallback(func(_ *glfw.Window, w, h int) {
```
Comment (hajimehoshi) [inline internal/uidriver/glfw/ui.go:826]: Please leave a comment explaining why this handling is needed.

**Comment by hajimehoshi** [comment]
Did you confirm this was not problematic on Windows and macOS?

**Comment by eliasdaler** [comment]
> Did you confirm this was not problematic on Windows and macOS?

Tested on Windows, works fine. Can't test on macOS unfortunately (don't have a machine running it)

File: internal/uidriver/glfw/ui.go [hunk 2]
```diff
@@ -821,7 +822,15 @@ event:
 			time.Sleep(time.Millisecond)
 		}
 	}
-	window.SetFramebufferSizeCallback(glfw.ToFramebufferSizeCallback(nil))
+	if u.defaultFramebufferSizeCallback == 0 {
+		// When window gets resized (either by manual window resize or a window
+		// manager), glfw sends a framebuffer size callback which we need to handle
```
Comment (hajimehoshi) [inline internal/uidriver/glfw/ui.go:827]: * Add the issue number.
* This problem is confirmed by i3 window manager but not by other managers, right?

So the comment should be like this:

```
		// When the window gets resized (either by manual window resize or a window
		// manager), glfw sends a framebuffer size callback which we need to handle (#1960).
		// This event is the only way to handle the size change at least on i3 window manager.
```

Comment (eliasdaler) [reply internal/uidriver/glfw/ui.go]: > This problem is confirmed by i3 window manager but not by other managers, right?

Yes, but I might try other tiling WMs, I'm just not used to other WMs. 
Also tested that xfwm (default XFCE window manager, not a tiling WM) works fine with this patch too

**Review by hajimehoshi — APPROVED** [review]
LGTM

**Pull request merged**